packages/backend-storage/src/construct.ts

```diff
@@ -148,13 +155,23 @@ export class AmplifyStorage
   private storeOutput = (
     outputStorageStrategy: BackendOutputStorageStrategy<StorageOutput> = new StackMetadataBackendOutputStorageStrategy(
       Stack.of(this)
-    )
+    ),
+    isDefault: boolean = false,
+    name: string = ''
   ): void => {
-    outputStorageStrategy.addBackendOutputEntry(storageOutputKey, {
+    /*
+     * The default bucket takes the `storageRegion` and `bucketName` name without a number post-fix.
+     */
+    outputStorageStrategy.appendToBackendOutputList(storageOutputKey, {
       version: '1',
       payload: {
-        storageRegion: Stack.of(this).region,
-        bucketName: this.resources.bucket.bucketName,
+        storageRegion: isDefault ? Stack.of(this).region : '',
+        bucketName: isDefault ? this.resources.bucket.bucketName : '',
+        buckets: JSON.stringify({
+          name,
+          bucketName: this.resources.bucket.bucketName,
+          storageRegion: Stack.of(this).region,
+        }),
```

Comment:
We should make sure that:
1. `storageRegion` and `bucketName` stay the same (not become array) and point to default bucket.
2. We append only to `buckets`. 

This may require some changes to outputStorageStrategy.
I think we should make `appendToBackendOutputList` to take `Partial<BackendOutputEntry>` to achieve that.

Reply:
Done 
3380bd9

`Partial` doesn't do "deep partial", so I had to introduce `PickPayload` and `TransformType`.